### PR TITLE
add `unstoppabledomains` to fetch process and graphql query

### LINKED ISSUE #81: Add `unstoppabledomains` data source
https://docs.unstoppabledomains.com/

Focus on domain resolve / ownership part.

### PR DESCRIPTION
**Pull request by ZhongFuze**
Add `unstoppabledomains` data source.
Resolves #81

### CODE REVIEW

File: src/upstream/rss3/mod.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@ use async_trait::async_trait;
 use chrono::{DateTime, NaiveDateTime};
 use futures::future::join_all;
 use http::uri::InvalidUri;
+use hyper::{Body, Client, Method, Request};
```
Comment (fengshanshan) [inline src/upstream/rss3/mod.rs:19]: these imports for ?

Comment (ZhongFuze) [reply src/upstream/rss3/mod.rs:19]: Some automatic imports, I'll delete them.

File: src/upstream/unstoppable/mod.rs [hunk 2]
```diff
@@ -0,0 +1,435 @@
...
+        updated_at: naive_now(),
+        fetcher: DataFetcher::RelationService,
+    };
+
+    let eth_record = eth_identity.create_or_update(&db).await?;
+    let unstoppable_record = unstoppable_identity.create_or_update(&db).await?;
+    hold.connect(&db, &eth_record, &unstoppable_record).await?;
+    let target = Target::Identity(
+        Platform::Ethereum,
+        result.meta.owner.clone().unwrap().to_lowercase(),
```
[the remark this answers is not in the document]
Comment (ZhongFuze) [reply src/upstream/unstoppable/mod.rs:380]: Yes, API will return structure but the value is null.
there are two situations that will occur:
1. Domain name has not been purchased or occupied
2. Domain name is purchased but not mint

[the remark this answers is not in the document]
Comment (ZhongFuze) [reply src/upstream/unstoppable/mod.rs:380]: So I I handle the `meta.owner.is_none()` case in the code above.

File: src/controller/graphql/resolve.rs [hunk 3]
```diff
@@ -113,10 +113,12 @@ impl ResolveQuery {
...
             }
-            DomainNameSystem::DotBit | DomainNameSystem::Lens => {
+            DomainNameSystem::DotBit | DomainNameSystem::Lens | DomainNameSystem::Unstoppable => {
                 let platform;
                 if domain_system == DomainNameSystem::DotBit {
                     platform = Platform::Dotbit;
+                } else if domain_system == DomainNameSystem::Unstoppable {
+                    platform = Platform::Unstoppable;
                 } else {
                     platform = Platform::Lens;
```
[the remark this answers is not in the document]
Comment (ZhongFuze) [reply src/controller/graphql/resolve.rs:123]: thx, it's more convenient to use `trait`

File: src/graph/edge/resolve.rs [hunk 4]
```diff
@@ -50,13 +50,31 @@ pub enum DomainNameSystem {
...
     #[strum(serialize = "unknown")]
     #[serde(rename = "unknown")]
     #[graphql(name = "unknown")]
     Unknown,
 }
 
+impl From<DomainNameSystem> for Platform {
+    fn from(domain: DomainNameSystem) -> Self {
+        match domain {
+            domain if domain == DomainNameSystem::DotBit => Platform::Dotbit,
```
Comment (nykma) [inline src/graph/edge/resolve.rs:69]: Why not just `DomainNameSystem::DotBit => Platform::Dotbit` ?

File: src/graph/edge/resolve.rs [hunk 5]
```diff
@@ -50,13 +50,31 @@ pub enum DomainNameSystem {
...
+            domain if domain == DomainNameSystem::Unstoppable => Platform::Unstoppable,
+            domain if domain == DomainNameSystem::Lens => Platform::Lens,
+            domain if domain == DomainNameSystem::Unknown => Platform::Unknown,
```
Comment (nykma) [inline src/graph/edge/resolve.rs:72]: No need to list `Unknown` separately.

**Review by nykma — CHANGES_REQUESTED** [review]
Need further modification.

File: src/controller/graphql/identity.rs [hunk 6]
```diff
@@ -152,7 +152,10 @@ impl IdentityRecord {
 
     /// there's only `platform: lens` identity `ownedBy` is not null
     async fn owned_by(&self, ctx: &Context<'_>) -> Result<Option<IdentityRecord>> {
-        if self.platform != Platform::Lens && self.platform != Platform::Dotbit {
+        if self.platform != Platform::Lens
+            && self.platform != Platform::Dotbit
+            && self.platform != Platform::Unstoppable
```
Comment (nykma) [inline src/controller/graphql/identity.rs:157]:
```suggestion

            if vec![...].contains(self.platform) {

```

**Review by nykma — CHANGES_REQUESTED** [review]
Need further modification.

File: src/controller/graphql/identity.rs (hunk 6 above)
Comment (ZhongFuze) [reply src/controller/graphql/identity.rs:157]: Oops!

File: src/controller/graphql/identity.rs [hunk 7]
```diff
@@ -152,10 +152,7 @@ impl IdentityRecord {
 
     /// there's only `platform: lens` identity `ownedBy` is not null
     async fn owned_by(&self, ctx: &Context<'_>) -> Result<Option<IdentityRecord>> {
-        if self.platform != Platform::Lens
-            && self.platform != Platform::Dotbit
-            && self.platform != Platform::Unstoppable
-        {
+        if vec![Platform::Lens, Platform::Dotbit, Platform::Unstoppable].contains(&self.platform) {
             return Ok(None);
         } else {
```
Comment (nykma) [inline src/controller/graphql/identity.rs:157]: Why putting `else` here since abnormal situation is already `return`ed above?

**Review by nykma — CHANGES_REQUESTED** [review]

**Review by nykma — APPROVED** [review]
:shipit:

**Pull request merged**